### PR TITLE
Verify software requirements in beginning of deployment scripts

### PR DESCRIPTION
**Pull request by avi-biton**
#59

### CODE REVIEW

File: deploy-deps.sh [hunk 1]
```diff
@@ -3,12 +3,48 @@
...
+    check_req
     echo "Deploying Konflux Dependencies" >&2
     deploy
     echo "Waiting for the dependencies to be ready" >&2
     "${script_path}/wait-for-all.sh"
 }
 
+check_req(){
+    # declare the requirements
+    requirements=(kind kubectl git openssl)
```
Comment (yftacherzog) [inline deploy-deps.sh]: I think we should not have kind here. This script runs after the part in which kind is involved. Also, I don't think we should couple this script to kind. 

About git, is it actually needed past this point? The scripts use it?

Maybe we should test that we access to kubeconfig and to a cluster. But maybe this is for another pr.

Comment (avi-biton) [reply deploy-deps.sh]: I was not so sure about `kind` but added it anyway.
We can decide what are the requirements we should check during the `deploy-deps.sh` script and edit it.
I think that for now it should be: `kubectl`, `openssl` and `podman` / `docker`

Comment (avi-biton) [reply deploy-deps.sh]: I removed `git` and `kind` from the requirements

File: deploy-deps.sh [hunk 2]
```diff
@@ -3,12 +3,48 @@
...
+    do
+        if ! command -v $i &> /dev/null; then
+                uninstalled_requirements+=("$i")
+        else
+                echo -e "$i is installed"
+    fi
+    done
+
+    # check if podman or docker are installed 
+    if ! command -v podman &> /dev/null && ! command -v docker &> /dev/null; then
```
Comment (gbenhaim) [inline deploy-deps.sh]: why do we need to check they are installed?

File: deploy-deps.sh [hunk 3]
```diff
@@ -3,12 +3,48 @@
...
+    uninstalled_requirements=()
+
+    # check if requirements are installed
+    for i in ${requirements[@]}
+    do
+        if ! command -v $i &> /dev/null; then
+                uninstalled_requirements+=("$i")
+        else
+                echo -e "$i is installed"
+    fi
```
Comment (gbenhaim) [inline deploy-deps.sh]: wrong indentation

File: deploy-deps.sh [hunk 4]
```diff
@@ -3,12 +3,48 @@
...
 }
 
+check_req(){
+    # declare the requirements
+    requirements=(kubectl openssl)
+    uninstalled_requirements=()
+
+    # check if requirements are installed
+    for i in ${requirements[@]}
+    do
```
Comment (gbenhaim) [inline deploy-deps.sh]: can we have the `do` in the same line with the for? (I wish we had a formmater for those things).

File: deploy-deps.sh [hunk 5]
```diff
@@ -3,12 +3,48 @@
...
+    check_req
     echo "Deploying Konflux Dependencies" >&2
     deploy
     echo "Waiting for the dependencies to be ready" >&2
     "${script_path}/wait-for-all.sh"
 }
 
+check_req(){
+    # declare the requirements
+    requirements=(kubectl openssl)
```
Comment (gbenhaim) [inline deploy-deps.sh]: please add local to the array declaration

File: deploy-deps.sh [hunk 6]
```diff
@@ -3,12 +3,48 @@
...
+    uninstalled_requirements=()
```
Comment (gbenhaim) [inline deploy-deps.sh]: please add local to the array declaration

File: deploy-deps.sh (hunk 2 above)
Comment (avi-biton) [reply deploy-deps.sh]: We checked it and when we are running the kind cluster creation we already need podman or docker, so the `deploy-deps.sh` should not check for it.
I will remove it

File: deploy-deps.sh (hunk 6 above)
Comment (avi-biton) [reply deploy-deps.sh]: Done

File: deploy-deps.sh (hunk 5 above)
Comment (avi-biton) [reply deploy-deps.sh]: Done

File: deploy-deps.sh (hunk 4 above)
Comment (avi-biton) [reply deploy-deps.sh]: Done

File: deploy-deps.sh (hunk 3 above)
Comment (avi-biton) [reply deploy-deps.sh]: Done

File: deploy-deps.sh (hunk 2 above)
Comment (avi-biton) [reply deploy-deps.sh]: Removed

**Review by yftacherzog — APPROVED** [review]

**Review by gbenhaim — APPROVED** [review]

**Pull request merged**